Sources/Formatter/FormatPatternValidator.swift

```diff
@@ -0,0 +1,50 @@
+//
+//  FormatPatternValidator.swift
+//  Linna
+//
+//  Created by Suita Fujino on 2018/11/30.
+//  Copyright © 2018 Suita Fujino. All rights reserved.
+//
+
+/**
+ A validator for the pattern of `CustomizableLogFormatter`.
+ */
+class FormatPatternValidator {
+    
+    public typealias ValidationResult = Result<Void, String>
+    
+    // MARK: - Properties
+    
+    let reservedParameters = ["%d", "%obj", "%level", "%file", "%func", "%line"]
+    
+    // MARK: - Methods
+    
+    /**
+     Vaildates the given pattern with replacing rules.
+     
+     - Parameters:
+       - pattern: String to validate.
+     
+     - Returns: `Result` with `Void` and the error message.
+     */
+    func validate(for pattern: String) -> Result<Void, String> {
+        if pattern.contains("%%") {
+            return .error(buildErrorMessage(characters: "%%"))
+        }
+        
+        var result = pattern
+        for param in reservedParameters {
+            result = result.replacingOccurrences(of: param, with: "A$")
+        }
+        if result.contains("%") {
+            // TODO: Specify the parameter
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/Scior/Linna/blob/0650b390e390e7340dbf846e3118908e66a5d168/Sources/Formatter/FormatPatternValidator.swift#L40"></span>TODOs should be resolved (Specify the parameter).</td>
    </tr>
  </tbody>
</table>